src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler{
+
+    @Override
+    public void handle(Storage storage, FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int quantityToBuy = fruitTransaction.getQuantity();
+        int currentQuantity = storage.fruits.getOrDefault(fruit, 0);
+
+        if (currentQuantity >= quantityToBuy) {
+            int newQuantity = currentQuantity - quantityToBuy;
+            storage.fruits.put(fruit, newQuantity);
+
+            if (newQuantity < 0) {
+                throw new RuntimeException("Incorrect quantity " + fruit + " quantity can't has negative value");
+            }

```

Logic issue: The code updates the storage with a potentially negative quantity before checking if the new quantity is negative. The check for negative quantity should occur before updating the storage to prevent invalid state. Also, the requirement might be to throw an exception if there is not enough stock, rather than just printing an error message. Please verify the checklist and requirements for correct error handling.